### PR TITLE
Fix video lock and format callback signatures to match libvlc

### PR DESCRIPTION
**Pull request by lummoj**
### Description of Change ###

Fixed callback signatures of LibVLCVideoLockCb and LibVLCVideoFormatCb to match libvlc and the comment blocks above.

IntPtr planes -> out IntPtr planes as it is [OUT]
IntPtr chroma -> ref uint chroma as it is [IN/OUT] and always 4 bytes (not platform dependent)

### Issues Resolved ### 
Fixes #219 (https://code.videolan.org/videolan/LibVLCSharp/issues/219)

### API Changes ###
Changed signature of LibVLCVideoLockCb and LibVLCVideoFormatCb to match libvlc

### Platforms Affected ### 
- Core (all platforms)

### Behavioral/Visual Changes ###
None

### Before/After Screenshots ### 
Not applicable

### Testing Procedure ###
Tested callbacks now work by rendering an I420 video to a RV32 (RGBA) image in WPF
More testing not likely to be required as the signatures are being fixed to match libvlc

### PR Checklist ###

- [*] Rebased on top of the target branch at time of PR
- [*] Changes adhere to coding standard

### CODE REVIEW

File: LibVLCSharp/Shared/MediaPlayer.cs [hunk 1]
```diff
@@ -1799,8 +1799,8 @@ public IntPtr GetFrame(out bool updated)
         /// <para>to not break assumptions that might be held by optimized code</para>
         /// <para>in the video decoders, video filters and/or video converters.</para>
         /// </remarks>
-        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
-        public delegate uint LibVLCVideoFormatCb(ref IntPtr userData, IntPtr chroma, ref uint width,
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl, CharSet = CharSet.Ansi)]
```
Comment (jeremyVignelles) [inline LibVLCSharp/Shared/MediaPlayer.cs]: Why did you change the CharSet?

Comment (lummoj) [reply LibVLCSharp/Shared/MediaPlayer.cs]: Thanks for the quick reply Jeremy, apologies CharSet should be reverted. I initially tried to get this working using string and char[] but I was not be able to get it work in conjunction with ref.

I'm happy to revert the CharSet change and I can see from the sample you linked how to get the lock callback working using Marshal.WriteIntPtr so perhaps this change is unnecessary.

Can this same approach be used for the chroma in the format callback? As I don't know the native width and height I'd prefer not to use SetVideoFormat as shown in the sample.

Comment (lummoj) [reply LibVLCSharp/Shared/MediaPlayer.cs]: I found a good example for chroma here and got everything working:
https://github.com/jeremyVignelles/Vlc.DotNet/blob/develop/src/Vlc.DotNet.Core.Interops/FourCCConverter.cs
Happy to abandon this now

**Comment by lummoj** [comment]
Found it was possible to achieve what I needed with existing callback signatures thanks to samples provided by @jeremyVignelles 

There is probably room for improvement with the API to make this more intuitive but I wouldn't consider it a bug.
